### PR TITLE
feat: update provider search styles

### PR DESCRIPTION
**Pull request by jacobvenable**
This PR does not close an issue.

## What does this PR do?

- implements new styles for the provider search
- creates new `<Autocomplete/>` which implements ARIA Autocomplete accessibility attributes based on [ARIA best practices](https://www.w3.org/TR/wai-aria-practices-1.1/examples/combobox/aria1.1pattern/listbox-combo.html)
- wraps provider search in new `<Autocomplete/>` component
- moves the search component towards the top of the main search providers page
- updates various use of "Resource" to "Provider"

## Screenshots

_hover via mouse styles_
hover

_focus via keyboard styles_
keyboard

## How does this PR make you feel? [:link:](http://giphy.com/categories/emotions/)

searching

### CODE REVIEW

File: packages/upswyng-server/src/components/Autocomplete.svelte [hunk 1]
```diff
@@ -0,0 +1,72 @@
+<script context="module">
+  export const key = {};
+</script>
+
+<script>
+  import { afterUpdate, onDestroy, setContext } from "svelte";
```
Comment (rhinodavid) [inline packages/upswyng-server/src/components/Autocomplete.svelte:6]: think you've got some unused imports here

File: packages/upswyng-server/src/components/ResourceSearch.svelte [hunk 2]
```diff
@@ -35,39 +43,37 @@
...
+    <div slot="help">
+      {#if hasQuery && !isLoading && !results.length}
+        <p class="help is-info">No providers match your search</p>
+      {/if}
+      {#if hasQuery && errorMessage}
+        <p class="help is-error">{errorMessage}</p>
+      {/if}
+    </div>
+    {#if hasQuery}
+      {#each results as resourceResult, i}
```
Comment (rhinodavid) [inline packages/upswyng-server/src/components/ResourceSearch.svelte:71]: probs should key this by the ID of the result.. I think as written is the default behavior but doesn't give you any benefits if the results change by the results length doesn't

File: packages/upswyng-server/src/routes/subcategory/[stub].svelte [hunk 3]
```diff
@@ -20,7 +20,7 @@
   let successMessage = "";
   let isSaving = false;
 
-  async function addResourceToSubcategory(resourceIdd /* string */) {
```
Comment (rhinodavid) [inline packages/upswyng-server/src/routes/subcategory/[stub].svelte:23]: thats fun

File: packages/upswyng-server/src/components/Autocomplete.svelte [hunk 4]
```diff
@@ -0,0 +1,72 @@
+<script context="module">
+  export const key = {};
```
Comment (rhinodavid) [inline packages/upswyng-server/src/components/Autocomplete.svelte:2]: I might be missing something -- is the key just an empty obj?

Comment (rhinodavid) [reply packages/upswyng-server/src/components/Autocomplete.svelte:2]: oh i saw the docs.. interesting

File: packages/upswyng-server/src/components/Autocomplete.svelte (hunk 1 above)
Comment (jacobvenable) [reply packages/upswyng-server/src/components/Autocomplete.svelte]: ahh, good call. [updated](https://github.com/CodeForBoulder/upswyng/pull/332/commits/cbedf4b2e7e6ce046ed2369ee71499a99df29885)

File: packages/upswyng-server/src/components/Autocomplete.svelte (hunk 4 above)
Comment (jacobvenable) [reply packages/upswyng-server/src/components/Autocomplete.svelte:2]: Ya I don't quite understand it myself but it appears to handle managing that itself.

File: packages/upswyng-server/src/components/ResourceSearch.svelte (hunk 2 above)
Comment (jacobvenable) [reply packages/upswyng-server/src/components/ResourceSearch.svelte]: 👍 [updated](https://github.com/CodeForBoulder/upswyng/pull/332/commits/391fc61f51613ecede33f4b95de61b35982ec4c8)

**Pull request merged**